feat: BROS-62: Responsive grid images

nick-skriabin opened the pull request:
This pull request updates the `GridView.scss` file to enhance the layout and styling of the `GridView` component. The changes introduce grid-based layout adjustments for better structure and responsiveness.

### Layout improvements:

* Added a new `&-body` class with a grid-based layout. This includes handling cases where the body contains an image, ensuring proper alignment and responsiveness. Additionally, images within the body are set to take up the full height.
* Updated the `&-content` class to use a grid layout with rows defined as `min-content` and `minmax(0, 1fr)`, improving the content structure and adaptability.

<img width="1279" alt="image" src="https://github.com/user-attachments/assets/44bac330-226f-42d2-a0a6-7bae93c6ffe2" />

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Make images responsive in grid view DM

yyassi-heartex reviewed (changes requested):
we need to be careful with this - what happens when a user adds more columns to display in grid view? since the image is now hight based does it just continuously get smaller? 
Alec mentioned `need to include an ability to change the size of the image / amount of text` and possibly the ability to change the size of the tile based on the number in the row so it gets bigger + the ability to scroll through text in the grid view?
Happy to sync on this with you - feels like there is more to be done here

tested in FB - I see the image can get squished in certain situations
[screen-recorder-wed-may-28-2025-13-36-47.webm](https://github.com/user-attachments/assets/e0b43c77-3248-444d-aa0f-10b7313308e5)

nick-skriabin committed:
Grid view settings

nick-skriabin committed:
Fix image max height

nick-skriabin committed:
Grouping same-type fields, cover text-only use case

nick-skriabin committed:
Fix calculation

nick-skriabin committed:
Fix cell height for text-only case

nick-skriabin committed:
Look of the button

Commit:
fix stuff, remove fit toggle in no-image scenario, min 1 image

Commit:
Add tests

niklub reviewed (changes requested):
Consider test coverage for `<GridView>`, which includes:
1. The main GridView component itself (in the current implementation, only GridBody was tested)
2. GridHeader component functionality
3. GridCell component interactions
4. GridDataGroup component behavior
5. Grid selection interactions (checkboxes, row selection)
6. Grid resizing and responsive behavior
7. Click handlers and event management
8. Different data types beyond images (Text, Unknown types)

Added autogenerated tests for your convenience, feel free to edit or remove them, but keeping the coverage high enough

nick-skriabin committed:
Tests

nick-skriabin committed:
Cleanup test

nick-skriabin committed:
UI fine tuning

yyassi-heartex commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15391371371)

yyassi-heartex committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15391371371

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15394479941)
> Successfully merged:  create mode 100644 web/libs/editor/src/components/TaskSummary/types.ts

hlomzik committed:
Merge branch 'develop' into 'fb-bros-62/dm-grid-images'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15394479941

nick-skriabin committed:
Update grid UI

nick-skriabin committed:
Merge branch 'develop' into fb-bros-62/dm-grid-images

hlomzik commented on web/libs/datamanager/src/components/DataManager/Toolbar/Toolbar.jsx:
```diff
@@ -11,6 +11,7 @@ const injector = inject(({ store }) => {
 
 export const Toolbar = injector(
   observer(({ store }) => {
+    console.log(store.SDK.toolbarInstruments);
```
```suggestion
```

hlomzik commented on web/libs/datamanager/src/components/MainView/GridView/GridView.jsx:
```diff
@@ -12,56 +12,87 @@ import { FF_GRID_PREVIEW, FF_LOPS_E_3, isFF } from "../../../utils/feature-flags
 import { GridViewContext, GridViewProvider } from "./GridPreview";
 import "./GridView.scss";
+import { clsx } from "clsx";
+clsx;
```
```suggestion
```

nick-skriabin committed:
Update web/libs/datamanager/src/components/MainView/GridView/GridView.jsx

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin committed:
Update web/libs/datamanager/src/components/DataManager/Toolbar/Toolbar.jsx

Co-authored-by: Andrew <hlomzik@gmail.com>

nick-skriabin committed:
Merge branch 'develop' into fb-bros-62/dm-grid-images

hlomzik commented on web/libs/datamanager/src/components/MainView/GridView/GridView.jsx:
```diff
@@ -92,14 +134,18 @@ export const GridView = observer(({ data, view, loadMore, fields, onChange, hidd
-      return res + height;
-    }, 16);
+          return res + height;
+        }, 16)
```
What is 16? Why do we add it to all heights?

nick-skriabin replied:
if only i remembered. this was here since forever and i don't remember why it's 16

hlomzik commented on web/libs/datamanager/src/components/MainView/GridView/GridView.jsx:
```diff
@@ -12,56 +12,87 @@ import { FF_GRID_PREVIEW, FF_LOPS_E_3, isFF } from "../../../utils/feature-flags
+import { groupBy } from "../../../utils/utils";
 
-const GridHeader = observer(({ row, selected }) => {
+const TEXT_ONLY_CELL_HEIGHT = 250;
```
this can be moved as a separate value to `TextDataGroup`, something like `TextDataGroup.heightWithoutImages = 250`

nick-skriabin replied:
it's actually incorrect name because it affects anything but images

hlomzik commented on web/libs/datamanager/src/components/MainView/GridView/GridView.jsx:
```diff
@@ -92,14 +133,18 @@ export const GridView = observer(({ data, view, loadMore, fields, onChange, hidd
+          return res + height;
+        }, 16)
+    : TEXT_ONLY_CELL_HEIGHT;
+  const finalRowHeight = (rowHeight + 32) * (hasImage ? Math.max(1, (7 - columnCount) * 0.5) : 1);
```
32 is a constant height of the header, it should not be modified by coefficient

hlomzik replied:
also better to make it a constant

nick-skriabin committed:
Rename constant to properly reflect the purpose

nick-skriabin committed:
Use header height as constant

yyassi-heartex commented on web/libs/datamanager/src/components/DataGroups/ImageDataGroup.jsx:
```diff
@@ -7,12 +7,14 @@ export const ImageDataGroup = (column) => {
+    columnCount,
   } = column;
   const root = getRoot(original);
+  const imageHeight = ImageDataGroup.height * Math.max(1, 8 - columnCount);
```
this is a little bit nit-picky but if we can define 8 as some sort of constant to help define what it is that would be great - it would help the code be self documenting and removes the magic number

hlomzik replied:
ah, yes, forgot to leave this comment! you are actually using 7 in another place, which is a very good example of issues without constanst like this

nick-skriabin replied:
✅

yyassi-heartex commented on web/libs/datamanager/src/components/DataManager/Toolbar/GridWidthButton.jsx:
```diff
@@ -1,49 +1,78 @@
+              <span>Fit images to width</span>
+              <Toggle
+                checked={!responsiveImage}
+                onChange={(e) => {
+                  view.setGridResponsiveImage(!e.target.checked);
+                }}
```
we should look at using a `useCallback` here

nick-skriabin replied:
✅

yyassi-heartex commented on web/libs/datamanager/src/sdk/dm-sdk.js:
```diff
@@ -176,6 +176,8 @@ export class DataManager {
       ...config.interfaces,
     });
 
+    console.log(this.toolbar);
+
```
```suggestion
```

hlomzik replied:
```suggestion
```

nick-skriabin committed:
Use `cnm` to properly merge TW classes

yyassi-heartex reviewed (approved):
approved with the expectation that the comments added by @hlomzik and I are addressed

nick-skriabin committed:
Update web/libs/datamanager/src/sdk/dm-sdk.js

Co-authored-by: Yousif Yassi <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Add `useCallback`

nick-skriabin committed:
Grid size coefficient constant

nick-skriabin commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15417103365)
> Successfully merged:  8 files changed, 104 insertions(+), 73 deletions(-)

nick-skriabin committed:
Merge branch 'develop' into 'fb-bros-62/dm-grid-images'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15417103365
